cryptoservice/crypto_service.go

```diff
@@ -58,16 +64,29 @@ func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (*dat
 	if err != nil {
 		return nil, fmt.Errorf("failed to add key to filestore: %v", err)
 	}
-	return data.PublicKeyFromPrivate(*privKey), nil
+	return data.PublicKeyFromPrivate(privKey), nil
+}
+
+// GetKey returns a key by ID
+func (ccs *CryptoService) GetKey(keyID string) data.Key {
+	key, err := ccs.keyStore.GetKey(keyID)
+	if err != nil {
+		return nil
+	}
+	return data.PublicKeyFromPrivate(key)
+}
+
+// DeleteKey deletes a key by ID
+func (ccs *CryptoService) DeleteKey(keyID string) error {
+	return ccs.keyStore.Remove(keyID)
 }
 
 // Sign returns the signatures for the payload with a set of keyIDs. It ignores
 // errors to sign and expects the called to validate if the number of returned
 // signatures is adequate.
 func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signature, error) {
-	// Create hasher and hash data
-	hash := crypto.SHA256
-	hashed := sha256.Sum256(payload)
+	var sha256Sum [sha256.Size]byte
+	sha256Computed := false
```

Comment:
I thought about that, but it would make ED25519 slower than it would be otherwise. If we expect people to use ED25519 in practice, I don't think we should compute a SHA256 sum and then throw it away. If we don't expect it to be used much in practice, that seems okay.